nav2_docking/opennav_docking/src/dock_database.cpp

```diff
@@ -203,7 +203,7 @@ bool DockDatabase::getDockInstances(const rclcpp_lifecycle::LifecycleNode::Share
       node->get_logger(), "Loading dock from database file  %s.", dock_filepath.c_str());
     try {
       return utils::parseDockFile(dock_filepath, node, dock_instances_);
-    } catch (YAML::ParserException & e) {
+    } catch (YAML::BadConversion & e) {
```

Comment:
Can you explain this one? `BadConversion`'s name implies that it is more specific than `ParserException` which I figure is better to cover all of the `YAML ` failure cases

Reply:
I had to change the ```ParserException``` to ```BadConversion``` because the ```utils::parseDockFile``` has its own try-catch when loading a file to catch the exceptions, here:

https://github.com/ros-navigation/navigation2/blob/5423eba732ed16b4f97a7353027c1f444b6edc60/nav2_docking/opennav_docking/include/opennav_docking/utils.hpp#L50-L55

The ```utils::parseDockFile``` function only throws a BadConversion if  the value is malformed, in this part:
https://github.com/ros-navigation/navigation2/blob/5423eba732ed16b4f97a7353027c1f444b6edc60/nav2_docking/opennav_docking/include/opennav_docking/utils.hpp#L67-L69

The ```ParserException``` is never thrown.